warrior/WarriorCore/Classes/html_results_class.py

```diff
@@ -9,432 +9,172 @@
-        name_list, value_list = [], []
-        arg_list = self._get_arglist(resultfile)
-        for arg in arg_list:
-            name = arg.get("name")
-            value = arg.get("value")
-            name_list.append(name)
-            value_list.append(value)
-        return name_list, value_list
-
-    def _get_arglist(self, resultfile):
-        """Get argument list from kw result file"""
-        args = xml_Utils.getChildElementWithSpecificXpath(resultfile, "./Keyword/Arguments")
-        args_list = args.findall("argument")
-        return args_list
+    def __init__(self, junit_file=None):
+        """ init function"""
+        self.junit_file = junit_file
+        self.html_template = "{0}{1}reporting{1}html_results_template.html" \
+            .format(Tools.__path__[0], os.sep)
+        self.junit_root = xml_Utils.getRoot(self.junit_file)
 
-    def _get_resultfile(self, junit_node, filetype, **kwargs):
-        """ get result file fro testcase/keywords"""
-        filepath = "None"
-        filename = junit_node.get("name")
-#         if filetype == "Keyword":
-#             tc_details = kwargs.get("tc_details", {})
-#             resultsdir = tc_details.get("tc_resultsdir", None)        
-#         else:
-            
-        if filetype == "Testcase":
-            resultsdir = junit_node.get("resultsdir")
-            filepath = str(resultsdir) + os.sep + "Results" + os.sep + str(filename) + "_results.xml"
-        elif filetype == "Keyword":
-            filepath = junit_node.get("resultfile")
-        return filepath
-        
-    def _get_icon(self, icon_type):
-        """Create a results icon"""
-        class_value = {"results":"fa fa-line-chart",
-                       "logs":"fa fa-book",
-                       "defects":"fa fa-bug"
-                       }.get(icon_type)
-        details = {"class":class_value}
-        results_icon = xml_Utils.create_element("i", **details)
-        return results_icon
-        
-    def _link_to_file(self, junit_node):
-        """Create a link to actual warrior file
-        based on the filetype"""
-        
-        filename = junit_node.get("name")
-        resultsdir = junit_node.get("resultsdir")
-        filepath = str(resultsdir) + os.sep + str(filename) + ".xml"
-        html_results_dir = os.path.dirname(self.html_results_path)
-        rel_filepath = os.path.relpath(filepath, html_results_dir)
-        href_link = xml_Utils.create_element("a", href=rel_filepath)
-        return href_link
+    def create_line_result(self, line, type):
+        """ create new objs"""
+        temp = LineResult()
+        temp.set_html(line, type, self.steps)
+        self.lineObjs.append(temp)
+        self.lineCount += 1
 
-    def _get_html_resultspath(self):
-        """Get the results path for the 
-        html results file"""
+    def set_line_objs(self):
+        """ call to create a new obj per item"""
+        self.lineCount = 0
+        project_node_list = [self.junit_root]
```

Comment:
is self.junit_root ever going to contain more that one ElementTree node at a time?